### DISCUSSION TITLE
Reconciliation loop i Docs example ?

### DISCUSSION THREAD

**Discussion by Kerwood** [kube-rs/kube#1463] [Q&A] [answered]
I'm learning kube.rs and I'm reading the docs for how to make a controller.
I was reading this section with an example of reconciliation https://kube.rs/controllers/reconciler/#in-depth-solution

What I'm getting from it is that the example is creating a Controller for a custom resource named `PodManager`, which will create a new `Pod` based on some information from the `PodManager` resource.

The code for the reconcile function is below.
```rs
let pod_data = create_owned_pod(&obj);
let serverside = PatchParams::apply("mycontroller");
let pod = pods.patch(pod.name_any(), serverside, Patch::Apply(pod_data)).await?

// update status object with the creation_timestamp of the owned Pod
let status = json!({
    "status": PodManagerStatus { pod_created: pod.meta().creation_timestamp }
});
api.patch_status(&obj.name_any(), &PatchParams::default(), &Patch::Merge(&status))
    .await?;
```

As you can see, the last step is to update the `PodManager` status field with `pod_created: <timestamp>`. But if the controller is created to watch `PodManager` resources, updating the status field will trigger a new reconciliation, a new timestamp will be set on the `PodManager` resource and the loop continues.

Am I missing something ?

**Comment by nightkr** [comment, marked as answer]
The `creation_timestamp` won't be updated when modifying a `Pod` (or other object) that already exists, only when creating a _new_ object.

**Reply by Kerwood** [reply]
Every time that code runs, a new timestamp will be updated on the `PodManager` resource, right ?

**Reply by Kerwood** [reply]
ooh `creation_timestamp`.. Got it. 
But the reconcile will run two times right ? When the `PodManager` gets the timestamp update the first time, the controller will run again, but this time the `patch_status()` will not update anything because the  `creation_timestamp` has not changed. 
Am I understanding it right ?

**Reply by nightkr** [reply]
Yes. Eventually the reconciliations should reach a stable state where no more modifications will happen and the loop stops.

**Reply by Kerwood** [reply]
Ok thanks.
